runtime/gc_check/CheckReporterTTY.cpp

```diff
@@ -196,15 +200,16 @@ GC_CheckReporterTTY::report(GC_CheckError *error)
 void 
 GC_CheckReporterTTY::reportObjectHeader(GC_CheckError *error, J9Object *objectPtr, const char *prefix)
 {
+	UDATA headerSize;
 	MM_GCExtensionsBase* extensions = MM_GCExtensions::getExtensions(_javaVM);
 	const char *prefixString = prefix ? prefix : "";
-	UDATA headerSize = extensions->objectModel.getHeaderSize(objectPtr);
 	PORT_ACCESS_FROM_PORT(_portLibrary);
 
-	if (!shouldReport(error)) {
+	if (!shouldReport(error) || NULL == J9GC_J9OBJECT_CLAZZ(objectPtr, extensions)) {
```

Comment:
Parentheses please for `==` subexpression.

Reply:
This has not been fixed.

Reply:
I am curious why this change is required. What is discovered as an object with NULL class in it (and why it was not required before)?

Reply:
It was a corner case that came up when testing a particular benchmark (I think it was the Hazelcast Jet Streaming benchmark). I can't recall the details exactly as I debugged this weeks ago, but I am sure this was required to handle the particular corner case.

Reply:
I think we should look closer to this case (not necessarily in this PR). In theory it should not happen. We should figure out why unexpected happened - is it defect of GC Check itself or there is a larger problem behind

Reply:
If it is possible to reproduce the problem and generate system core for it (using `abort` sub option for example) I can help to figure out what happened.